examples/goldilocks_models/find_boundary_utils/plot_landscape.py

```diff
@@ -0,0 +1,141 @@
+           '_large_iter150/'
+    dir5 = file_dir+'dairlib_data/find_boundary/' + robot + '2D_rom/2D_task_space/' + 'robot_' + str(robot_option) + \
+           '_2d_initial_model/'
+    dir6 = file_dir+'dairlib_data/find_boundary/' + robot + '2D_rom/2D_task_space/' + 'robot_' + str(robot_option) + \
+           '_small_iter200/'
+    dir7 = file_dir+'dairlib_data/find_boundary/' + robot + '2D_rom/2D_task_space/' + 'robot_' + str(robot_option) + \
+           '_large_iter100/'
+
+
+i = 0
+x = []
+y = []
+z = []
+dir = dir4
+dir_nominal = dir3
+found_same_sample = False
+while os.path.isfile(dir+str(i)+'_'+str(0)+'_gamma.csv'):
+    gamma = np.genfromtxt(dir + str(i) + '_' + str(0) + '_gamma.csv', delimiter=",")
+    cost = float(np.genfromtxt(dir + str(i) + '_' + str(0) + '_c.csv', delimiter=","))
+    if np.genfromtxt(dir + str(i) + '_' + str(0) + '_is_success.csv', delimiter=",") == 1:
+        if normalized_cost:
+            j = 0
+            # search the nominal cost map
+            while os.path.isfile(dir_nominal+str(j)+'_'+str(0)+'_gamma.csv'):
+                gamma_nominal = np.genfromtxt(dir_nominal + str(j) + '_' + str(0) + '_gamma.csv', delimiter=",")
+                if np.array_equal(gamma, gamma_nominal):
+                    found_same_sample = True
+                    if np.genfromtxt(dir_nominal + str(j) + '_' + str(0) + '_is_success.csv', delimiter=",") == 1:
+                        nominal_cost = float(np.genfromtxt(dir_nominal + str(j) + '_' + str(0) + '_c.csv', delimiter=","))
+                        if cost/nominal_cost < 1.5:
+                            x.append(gamma[0])
+                            y.append(gamma[1])
+                            z.append(cost/nominal_cost)
+                    break
+                j = j+1
+            if found_same_sample:
+                found_same_sample = False
+            else:
+                x.append(gamma[0])
+                y.append(gamma[1])
+                z.append(0)
+        else:
+            x.append(gamma[0])
+            y.append(gamma[1])
+            z.append(cost)
+    i = i+1
+
+fig, ax = plt.subplots()
+
+print(np.shape(x))
+print(np.shape(y))
+print(np.shape(z))
+
+# continuous color map
+surf = ax.tricontourf(x, y, z)
+fig.colorbar(surf, shrink=0.5, aspect=6)
+ax.set_xlabel('Stride length')
+ax.set_ylabel('Ground incline')
+if normalized_cost:
+    ax.set_title('normalized cost landscape')
+else:
+    ax.set_title('cost landscape')
+
+# optimization range
+# large range
+if plot_large_range == 1:
+    x_line = np.array([min_sl1, max_sl1, max_sl1, min_sl1, min_sl1])
+    y_line = np.array([min_gi1, min_gi1, max_gi1, max_gi1, min_gi1])
+    ax.plot(x_line, y_line, 'black', linewidth=3)
+# small range
+if plot_small_range == 1:
+    x_line2 = np.array([min_sl2, max_sl2, max_sl2, min_sl2, min_sl2])
+    y_line2 = np.array([min_gi2, min_gi2, max_gi2, max_gi2, min_gi2])
+    ax.plot(x_line2, y_line2, 'black', linewidth=3)
```

Comment:
It looks like we are plotting two boxes here? Shouldn't we only plot one box since we are not comparing two cost landscape?

Reply:
We can choose which box to plot by setting the falg `plot_xxx_range`. This function is also used for comparing the results of optimizing over different task space (Eg. normalizing the cost from the large one with the cost from the small one).